tools/ml-audio-content-moderation/terraform/main.tf

```diff
@@ -0,0 +1,253 @@
+resource "google_storage_bucket" "output_bucket" {
+  provider = "google"
+  name = "output-files-${substr(lower(random_uuid.uuid.result), 0, 20)}"
+}
+
+
+# Create PubSub resources
+resource "google_pubsub_topic" "stt_topic" {
+  name = "${var.stt_queue_topic_name}"
+  depends_on = ["google_project_service.pubsubapi"]
+}
+
+resource "google_pubsub_subscription" "pull_stt_ids" {
+  name = "${var.stt_queue_subscription_name}"
+  depends_on = ["google_project_service.pubsubapi"]
+  topic = "${google_pubsub_topic.stt_topic.name}"
+}
+
+
+# Zip up NLP API source code folder
+data "archive_file" "nlp_api_function" {
+  type = "zip"
+  output_path = "nlp.zip"
+  source_dir = "../nlp_api_function"
+}
+
+# Zip up Perspective API source code folder
+data "archive_file" "perspective_api_function" {
+  type = "zip"
+  output_path = "perspective.zip"
+  source_dir = "../perspective_api_function"
+}
+
+# Zip up Read STT source code folder
+data "archive_file" "read_stt" {
+  type = "zip"
+  output_path = "read_stt.zip"
+  source_dir = "../read_stt_api_function"
+}
+
+# Zip up Send STT source code folder
+data "archive_file" "send_stt" {
+  type = "zip"
+  output_path = "send_stt.zip"
+  source_dir = "../send_stt_api_function"
+}
+
+# Store STT source code
+resource "google_storage_bucket" "function_source_code" {
+  name = "source-code-${substr(lower(random_uuid.uuid.result), 0, 20)}"
+}
+
+resource "google_storage_bucket_object" "send_stt_code" {
+  name = "send_stt_api_source"
+  bucket = "${google_storage_bucket.function_source_code.name}"
+  source = "send_stt.zip"
+}
+
+resource "google_storage_bucket_object" "read_stt_code" {
+  name = "read_stt_api_source"
+  bucket = "${google_storage_bucket.function_source_code.name}"
+  source = "read_stt.zip"
+}
+
+
+resource "google_storage_bucket_object" "perspective_code" {
+  name = "perspective_api_source"
+  bucket = "${google_storage_bucket.function_source_code.name}"
+  source = "perspective.zip"
+}
+
+
+resource "google_storage_bucket_object" "nlp_code" {
+  name = "nlp_api_source"
+  bucket = "${google_storage_bucket.function_source_code.name}"
+  source = "nlp.zip"
+}
+
+
+resource "google_cloudfunctions_function" "send_stt_api" {
+  depends_on = ["google_project_service.speechapi", "google_pubsub_topic.stt_topic"]
+  name = "send_stt_api"
+  region = "${var.cloud_functions_region}"
+  entry_point = "main"
+  runtime = "python37"
+  environment_variables = {
+    topic_name = "${google_pubsub_topic.stt_topic.name}"
+    error_audio_bucket = "${google_storage_bucket.error_audio_bucket.name}"
+
+  }
+  source_archive_bucket = "${google_storage_bucket.function_source_code.name}"
+  source_archive_object = "${google_storage_bucket_object.send_stt_code.name}"
+  timeout = "540"
+  event_trigger {
+    resource = "${google_storage_bucket.staging_audio_bucket.name}"
+    event_type = "google.storage.object.finalize"
+  }
+}
+
+resource "google_cloudfunctions_function" "read_stt_api" {
```

Comment:
You could simplify this using this module: https://github.com/terraform-google-modules/terraform-google-scheduled-function

Reply:
Same as comment above.